lib/tasks/search.rake

```diff
@@ -4,6 +4,21 @@ namespace :search do
     Document.find_by(content_id: args[:content_id]).published_edition.update_in_search_index
   end
 
+  desc "Re-index a collection of Documents with specified world location. Takes a `world_location_slug` as argument."
+  task :resend_document, [:world_location_slug] => [:environment] do |_, args|
+    world_location = WorldLocation.find_by(slug: args[:world_location_slug])
+
+    if world_location
+      content_ids = Edition.in_world_location(world_location.id).map(&:content_id)
+
+      Document.where(content_id: content_ids).each do |document|
+        document.published_edition.update_in_search_index
```

Comment:
To avoid having to do `published_edition` here (and reduce the number of queries), I wonder if you could change the query to something like this:

```rb
documents = Edition.in_world_location(world_location.id).where(state: Edition::PUBLICLY_VISIBLE_STATES).map(&:document)
documents.each do { |d| d.update_in_search_index }
```

Reply:
That is not exactly the same:
```
irb(main):009:0> Document.where(content_id: Edition.in_world_location(25).map(&:content_id)).each(&:published_edition).count
=> 424
irb(main):010:0> Edition.in_world_location(25).where(state: Edition::PUBLICLY_VISIBLE_STATES).map(&:document).count
=> 413
```